src/lib/containers/Markdown.js

```diff
@@ -0,0 +1,481 @@
+     *
+     * @param {*} widgetType    type of widget, e.g. "buttonlink"
+     * @param {*} widgetparamsMapped    parameter dictionary for the given widget
+     * @param {*} element   reference to the actual DOM element this widget corresponds to
+     * @param {*} fileHandleAssociationList  stack of of requests to be made to batch synapse request
+     * @param {*} elementList   stack of elements to be processed
+     */
+    prepareWidget(widgetType, widgetparamsMapped, element, fileHandleAssociationList, elementList) {
+        if (widgetType === "buttonlink") {
+            let button = "<a href=\"" + widgetparamsMapped.url + "\"class=\"btn btn-lg btn-info\" role=\"button\" >" + widgetparamsMapped.text + "</a>";
+            element.outerHTML = button;
+        } else if (widgetType === "image" && this.state.fileHandles) {
+            let fileName = decodeURIComponent(widgetparamsMapped.fileName);
+            let match = this.matchToHandle(this.compareById(fileName, "fileName"), this.state.fileHandles.list);
+            if (match.length > 0) {
+                fileHandleAssociationList.push({
+                    fileHandleId: match[0].id,
+                    associateObjectId: this.state.wikiId,
+                    associateObjectType: "WikiAttachment"
+                });
+                elementList.push({element:element, id: match[0].id, widgetType: widgetType, widgetparamsMapped: widgetparamsMapped});
+            }
+        } else if (widgetType === "plot") {
+            elementList.push({element:element, widgetType: widgetType, widgetparamsMapped: widgetparamsMapped});
+        }
+    }
+
+    /**
+     * match all widgets on the page to their corresponding resource
+     *
+     * @param {*} elementList dictionary of 
+     * @memberof Markdown
+     */
+    matchElementToResource(elementList) {
+        elementList.forEach(elementBundle => {
+            let match = this.matchToHandle(this.compareById(elementBundle.id, "fileHandleId"), this.state.fileResults);
+            let renderedHTML = ""
+            if (elementBundle.widgetType === "image") {
+                renderedHTML = "<image class=\"img-fluid\" src=" + match[0].preSignedURL + "></image>";
+                elementBundle.element.outerHTML = renderedHTML
+            } else if (elementBundle.widgetType === "plot") {
+                let widgetparamsMapped = elementBundle.widgetparamsMapped 
+                let raw_plot_data = {}
+
+                if (!this.state.queryData) {
+                    // grab all the data, hasn't been loaded yet
+                    let queryRequest = {
+                        // TODO: verify these parameters
+                        concreteType: "org.sagebionetworks.repo.model.table.QueryBundleRequest",
+                        entityId: this.state.ownerId,
+                        partsMask: 13,
+                        query: {
+                            isConsistent: false,
+                            limit: 150,
```

Comment:
Put into loop.  Initial request should ask for max page size.  Following requests should use that max page size as the limit (and update the offset for each page!), and quit when result size is less than limit.